dali/python/nvidia/dali/pipeline.py

```diff
@@ -711,67 +762,21 @@ def build(self, define_graph = None):
         self._pipe.Build(self._names_and_devices)
         self._built = True
 
-    def _feed_input(self, name, data, layout = None, cuda_stream = None, use_copy_kernel = False):
-        from nvidia.dali.external_source import _check_data_batch
-
-        infer_stream = False
+    def _feed_input(self, name, data, layout=None, cuda_stream=None, use_copy_kernel=False):
         if cuda_stream is None:
-            infer_stream = True
+            cuda_stream = _get_default_stream_for_array(data)
```

Comment:
I noticed that before when data was a list we called `_get_default_stream_for_array` multiple times. Was there any particular reason for that?